cmd/bugzilla-backporter/main.go

```diff
@@ -0,0 +1,187 @@
+	validateOnly bool
+	flatRegistry bool
+}
+
+type traceResponseWriter struct {
+	http.ResponseWriter
+	statusCode int
+	size       int
+}
+
+func handleWithMetrics(h http.HandlerFunc) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		t := time.Now()
+		// Initialize the status to 200 in case WriteHeader is not called
+		trw := &traceResponseWriter{ResponseWriter: w, statusCode: http.StatusOK}
+		h(trw, r)
+		latency := time.Since(t)
+		labels := prometheus.Labels{"status": strconv.Itoa(trw.statusCode), "path": r.URL.EscapedPath()}
+		bzbpMetrics.httpRequestDuration.With(labels).Observe(latency.Seconds())
+		bzbpMetrics.httpResponseSize.With(labels).Observe(float64(trw.size))
+	})
+}
+
+// Needs additional work
+func getAuthToken() []byte {
+	// Auth token commented out for security reasons
+	// Will modify function to use serviceAgent
+	return []byte("")
+}
+
+// Handler returns a function which populates the response with the details of the bug
+// Expects a POST request with field ID
+// Returns bug details in JSON format
+func getBugHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			w.WriteHeader(http.StatusNotImplemented)
+			w.Write([]byte(http.StatusText(http.StatusNotImplemented)))
+			return
+		}
+
+		encoded, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("Could not read bug ID from request body."))
+			return
+		}
+		client := bugzilla.NewClient(getAuthToken, "https://bugzilla.redhat.com/")
+		// Reusing the bug struct from bugzilla namespace
+		bug := bugzilla.Bug{}
+		if err = json.Unmarshal(encoded, &bug); err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("Could not parse request body as unresolved config."))
+			return
+		}
+		bugInfo, err := client.GetBug(bug.ID)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("Bug ID not found"))
+			return
+		}
+		jsonBugInfo, err := json.MarshalIndent(*bugInfo, "", "  ")
+		if err != nil {
+			recordError("failed to marshal bugInfo")
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "failed to marshal bugInfo to JSON: %v", err)
+			// logger.WithError(err).Errorf("failed to marshal bugInfo to JSON")
+			return
+		}
+		fmt.Println(bugInfo)
+		w.WriteHeader(http.StatusOK)
+		w.Write(jsonBugInfo)
+	}
+}
+
+func genericHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusNotFound)
+		w.Write([]byte(http.StatusText(http.StatusNotFound)))
+	}
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.logLevel, "log-level", "info", "Level at which to log output.")
+	fs.StringVar(&o.address, "address", ":8080", "Address to run server on")
+	fs.StringVar(&o.uiAddress, "ui-address", ":8082", "Address to run the registry UI on")
```

Comment:
i am expecting to use it in the next phase, for the front end server

Reply:
Okay, I think it would be easier to just have one listener with multiple endpoints, the reason this ended up being split in in the ci-operator-configresolver is that one of the listeners uses plain http and one uses tls.

Also, as a bit of a heads-up, we want the front-end to be super simplistic and just a go template that renders some html/css and no javascript/typescript, check as a sample e.g. here: https://github.com/openshift/ci-tools/blob/master/pkg/webreg/webreg.go#L1895

The reason for this is that we do not have anyone on the team who has experience to speak of in javascript/typescript and by sticking to go, everyone can make and review changes. 

We can rethink this decision should we end up needing something dynamic, but I think it should be fine without.

Reply:
Alright sounds good, thank you for the clarification.